packages/near-membrane-dom/src/window.ts

```diff
@@ -71,12 +71,14 @@ export function getCachedGlobalObjectReferences(
     const EventTargetProto = ReflectGetPrototypeOf(WindowPropertiesProto)!;
     record = {
         document,
-        DocumentProto: ReflectGetPrototypeOf(document)!,
+        DocumentProto: SUPPORTS_SHADOW_REALM ? undefined : ReflectGetPrototypeOf(document)!,
```

Comment:
this should always exists, and can always be extracted for now... yes, eventually we can add optimizations to not inspect a bunch of things that we don't care anymore when running in shadowRealm, but for now, I think we should avoid this forking logic since this code runs only once per window instance.

Reply:
Ya that's a minor tweak. Once we figure out why window prototype isn't inheriting the rest can better fall into place.

Reply:
Okay 😸 . Once we figure out why window prototype isn't inheriting the rest can better fall into place. I'm hoping you can figure it out as it doesn't seem to make much sense (maybe a bug in the ShadowRealm implementation?)